src/java/org/apache/cassandra/db/commitlog/CommitLogSegment.java

```diff
@@ -173,7 +177,12 @@ static long getNextId()
 
         try
         {
-            channel = FileChannel.open(logFile.toPath(), StandardOpenOption.WRITE, StandardOpenOption.READ, StandardOpenOption.CREATE);
+            // We need both READ and WRITE for Memory mapped segments (there is no write only shared mapping) but
+            // some storage type doesn't support mmapped segments
+            if (DatabaseDescriptor.getDiskAccessMode() == Config.DiskAccessMode.standard)
```

Comment:
nit: 
it would be in line with the comment to check against mmap, not standard, wouldn't it?

Reply:
There are multiple mmap modes, and just the one that isn't.